enhancements/ingress/custom-route-configuration.md

```diff
@@ -72,82 +76,97 @@ slightly different documentation.
 type IngressSpec struct {
     // other fields
 
-    // CustomizeableRoutes is a list of routes that a cluster-admin wants to customize.  It is logically keyed by
-    // .spec.customizeableRoutes[index].{namespace,name}.
-    // To determine the set of possible keys, look at .status.customizeableRoutes where participating operators place
+    // ComponentRoutes is a list of routes that a cluster-admin wants to customize.  It is logically keyed by
+    // .spec.componentRoutes[index].{namespace,name}.
+    // To determine the set of possible keys, look at .status.componentRoutes where participating operators place
     // current route status keyed the same way.
-    CustomizeableRoutes []CustomizeableRouteSpec
+    // If a ComponentRoute is created with a namespace,name tuple that does not match status, that piece of config will
+    // not have an effect.  If an operator later reads the field, it will eventually (but not necessarily immediately)
+    // honor the pre-existing spec values.
+    ComponentRoutes []ComponentRouteSpec
 }
 
 type IngressStatus struct {
     // other fields
 
-    // CustomizeableRoutes is where participating operators place the current route status for routes which the cluster-admin
+    // ComponentRoutes is where participating operators place the current route status for routes which the cluster-admin
     // can customize hostnames and serving certificates.
     // How the operator uses that serving certificate is up to the individual operator.
-    CustomizeableRoutes []CustomizeableRouteStatus
+    // An operator that creates entries in this slice should clean them up during removal (if it can be removed).
+    // An operator must also handle the case of deleted status without churn.
+    ComponentRoutes []ComponentRouteStatus
 }
 
-type CustomizeableRouteSpec struct{
-    // namespace is the namespace of the route to customize.  It must be a real namespace.
+type ComponentRouteSpec struct{
+    // namespace is the namespace of the route to customize.  It must be a real namespace.  Using an actual namespace
+    // ensures that no two components will conflict and the same component can be installed multiple times.
     Namespace string
     // name is the *logical* name of the route to customize.  It does not have to be the actual name of a route resource.
     // Keep in mind that this is your API for users to customize.  You could later rename the route, but you cannot rename
     // this name.
     Name string
-    // desiredHost is the name that a cluster-admin wants to specify
-    DesiredHost string
+    // Hostname is the host name that a cluster-admin wants to specify
+    Hostname string
```

Comment:
still open question: what if there are multiple?

Reply:
> still open question: what if there are multiple?

Routes can only specify a single name (see https://github.com/openshift/api/blob/master/route/v1/types.go#L80).  I don't understand the question.  Are you proposing an API that coerces operators into the creation of multiple routes?

Reply:
i thought about the same and (sorry for xrefing) as commented here https://github.com/openshift/enhancements/pull/577#discussion_r565146419 a 1:1 relationship between a `Route` and `CustomizeableRoute` seems reasonable.

If the user wants multiple hostnames for a given use case (i.e. multiple external hostnames for an externally visible Prometheus), the underlying operator a) would need to support creating multiple routes (one for each hostname) and b) support for allowing customizing each of them. This is equivalent to what the user (to my knowledge) needs to do, namely create seperate one route per hostname.

Does that sound reasonable?